server/container/core/src/main/java/org/apache/james/server/core/MailImpl.java

```diff
@@ -143,10 +147,12 @@ public static Builder builder() {
         private Optional<String> remoteHost;
         private PerRecipientHeaders perRecipientHeaders;
 
-        private Builder() {
+        private Builder(String name) {
+            Preconditions.checkNotNull(name);
+            Preconditions.checkArgument(!name.isEmpty());
```

Comment:
you don't add a comment?

Reply:
and it could be simplified by `StringUtils.isNotBlank()`

Reply:
> you don't add a comment?

About nullity ?

Reply:
> and it could be simplified by `StringUtils.isNotBlank()`

See my previous answer, that's not the case because I want two different exceptions

Reply:
> About nullity ?

No, about emptyness.
```suggestion
            Preconditions.checkArgument(!name.isEmpty(), "name must not be empty");
```

Reply:
fixed